### ISSUE TITLE
RequestDelegateGenerator requires the object to implement `IParsable<T>` or otherwise have a `TryParse` method.

### ISSUE THREAD

**Issue by KennethHoff** [dotnet/aspnetcore#54277]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

With the `RequestDelegateGenerator` enabled (`PublishAoT` or `EnableRequestDelegateGenerator` to `true`), minimal api endpoints with `[FromForm]` on a model without a custom-made `TryParse` method will not compile correctly.

`[FromBody]` - which I imagine would work more-or-less the same in this regard - does not have this issue.

### Expected Behavior

`[FromForm]` and `[FromBody]` should work identically, and should not require the developer to create their own `TryParse` method.

### Steps To Reproduce

https://github.com/KennethHoff/Repros/tree/66a2d8c3b53fd64dfc5f3880cb2db8d9cac6550a/AspNetCoreRequestDelegateGenerator

### Exceptions (if any)

```
0>GeneratedRouteBuilderExtensions.g.cs(520,59): Error CS0029 : Cannot implicitly convert type 'string' to 'NotParsableForm'
0>GeneratedRouteBuilderExtensions.g.cs(551,59): Error CS0029 : Cannot implicitly convert type 'string' to 'NotParsableForm'
0>GeneratedRouteBuilderExtensions.g.cs(637,67): Error CS0029 : Cannot implicitly convert type 'string' to 'ImplicitlyParseOnlyForm'
0>GeneratedRouteBuilderExtensions.g.cs(668,67): Error CS0029 : Cannot implicitly convert type 'string' to 'ImplicitlyParseOnlyForm'
```

### .NET Version

`8.0.100`, `8.0.201`, `9.0.100-preview.1.24101.2`

**Comment by captainsafia** [comment]
@KennethHoff The error that you're running into is a result of the difference in behavior when trying to bind _complex_ types from a form. A history of the behavior here:

- In. NET 7, we didn't support the source generator and attempting to bind complex types from from a form. We assumed that everything you applied `[FromForm]` on would be parsable. If it wasn't, you ran into the exception you saw here.
- In .NET 8, we attend support for binding complex type but _only_ for non-AoT/RDG scenarios. You''ll get the same exception when running with RDG on.

If you'd like complex form binding support, I'd recommend you turn off RDG. If you require native AoT support for your scenario, let us know.

The error message could be better here but we don't do anything to capture the exception throw and provide a more detailed log.

Does disabling RDG work for supporting this particular case?

**Comment by KennethHoff** [comment]
@captainsafia I only enabled RDG because I figured it would just be a better performing, easier-to-debug/step-through variant of the RDF.

.Net 8 added support for `[FromForm]` in Minimal APIs as well as support for RDG, so I simply put two and two together and figured we could use `[FromForm]` with the RDG.

The error message being so terrible is what made me assume that this was a bug rather than a feature.

I have no real need for it, but I _want to_ use it! 😅. I also don't really see a major reason (read: other than the standard "hasn't been done yet") why this wouldn't be a supported scenario.

Is it something on your radar? Could it conceivably be fixed in Asp.Net Core 9?

**Comment by KennethHoff** [comment]
At the very least, I believe an improved error message is in order, as it's gives off very "broken functionality" vibes 🫠

**Comment by captainsafia** [comment]
> Is it something on your radar? Could it conceivably be fixed in Asp.Net Core 9?

Yep, it's on our radar. For context, we integrated complex form binding support in minimal APIs by doing code sharing with the form binding that was supported in Blazor. We did this to have consistency between the two stacks and to reduce the implementation cost. However, the code we source-share with Blazor for this makes heavy use of dynamic code-gen and reflection. For us to resolve this issue, we'd need to update the shared source which is a bit of an undertaking.

But yes, agree that the error message could definitely be better here 😅

**Comment by KennethHoff** [comment]
So, what I'm hearing here is that in order for Blazor to support AoT, this has to be fixed, and I'm guessing that's near the top of your priorities 🤞🏻

(Again, this does not really affect me; it's mostly just cool to say I have a source generated "[endpoint-]handler handler")

**Comment by KennethHoff** [comment]
>Full disclosure: It's likely not happening for .NET 9 based on the Blazor roadmap (https://github.com/dotnet/aspnetcore/issues/51834). 😅

No worries. As previously mentioned; This does not block anything on my end :)
